src/simulator/map.py

```diff
@@ -44,8 +45,13 @@ def instantiate_object(self, obj_json, new_id: ObjectId, name: string, owner: Ob
             # TODO also skipping name for now
             return RigidPointBall(new_id, shape, position, 0, 2, owner, self.registry)
         elif cname == "VacuumCleanerV0":
+            bases = obj_json["bases"]
             return VacuumCleanerV0(new_id, name, position, owner, {"diameter": shape.radius, "height": shape.height},
```

Comment:
are you only passing the object names? how does the robot have access to the actual base objects then?